src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -176,28 +230,32 @@ private SwiftCompileStep makeCompileStep() {
               public String apply(SourcePath input) {
                 return getResolver().getAbsolutePath(input).toString();
               }
-            })));
+            })
+            .toSet()));
 
     Optional<Iterable<String>> configFlags = swiftBuckConfig.getFlags();
     if (configFlags.isPresent()) {
       compilerCommand.addAll(configFlags.get());
     }
-    compilerCommand.add(
-        "-enable-testing",
-        "-c",
-        enableObjcInterop ? "-enable-objc-interop" : "",
-        hasMainEntry ? "" : "-parse-as-library",
-        "-module-name",
-        moduleName,
-        "-emit-module",
-        "-emit-module-path",
-        modulePath.toString(),
-        "-o",
-        objectPath.toString(),
-        "-emit-objc-header-path",
-        headerPath.toString());
-    for (SourcePath sourcePath : srcs) {
-      compilerCommand.add(getResolver().getRelativePath(sourcePath).toString());
+
+    if (bridgingHeader.isPresent()) {
+      compilerCommand.add(
+          "-import-objc-header",
+          getResolver().getRelativePath(bridgingHeader.get()).toString());
+    }
+
+    if (src.isPresent()) {
```

Comment:
It's really strange to have the rule behave so differently based on whether it has a source or not, can you split this into two rules?